src/filtering/filterInput.component.tsx

```diff
@@ -62,8 +62,82 @@ const FilterInput = (props: FilterInputProps) => {
         setError('');
         setInputIndex((prevIndex) => prevIndex + -1);
       }
+      // allow selecting operators, channels, strings
+      // and numbers with Space key down
+      if (e.key === ' ') {
+        const operatorExactMatch = operators.find(
+          (opt) => opt.value === inputValue
+        );
+        const channel = channels.filter((channel) =>
+          channel.label.toLowerCase().startsWith(inputValue.toLowerCase())
+        );
+        const operatorListMatch = operators.filter((opt) =>
+          opt.value.toLowerCase().startsWith(inputValue.toLowerCase())
+        );
+
+        const newValue = [...value];
+        let newToken;
+
+        switch (true) {
+          case channel.length === 1:
+            newToken = channel[0];
+            break;
+
+          case operatorExactMatch !== undefined:
+            newToken = operatorExactMatch;
+            break;
+
+          case operatorListMatch.length === 1:
+            newToken = operatorListMatch[0];
+            break;
+
+          case !Number.isNaN(Number(inputValue)) &&
+            inputValue.trim().length > 0:
+            newToken = {
+              type: 'number',
+              value: inputValue,
+              label: inputValue,
+            };
+            break;
+
+          case (inputValue[0] === '"' &&
+            inputValue[inputValue.length - 1] === '"') ||
+            (inputValue[0] === "'" &&
+              inputValue[inputValue.length - 1] === "'"):
+            newToken = {
+              type: 'string',
+              value: inputValue,
+              label: inputValue,
+            };
+            break;
+
+          default:
+            break;
+        }
+
+        if (
+          (channel.length === 1 ||
+            operatorExactMatch !== undefined ||
+            operatorListMatch.length === 1 ||
+            (!Number.isNaN(Number(inputValue)) &&
+              inputValue.trim().length > 0) ||
+            (inputValue[0] === '"' &&
+              inputValue[inputValue.length - 1] === '"') ||
+            (inputValue[0] === "'" &&
+              inputValue[inputValue.length - 1] === "'")) &&
+          newToken
```

Comment:
Can't we just test if `typeof newToken !== undefined`? As `newToken` by definition is only defined when one of the cases in the switch statement matches. Means we don't have to repeat the case match logic here

Reply:
I've done the changes below. I made one change to use  `newToken !== undefined` as the type of newToken is `Token | undefined` and using `typeof newToken !== undefined` causes error as it true when newToken is undefined

Reply:
Ah yeah, I made a typo - the typeof version of the check needs to be `typeof newToken !== "undefined"`. However, that check is just an extra safe version of `newToken !== undefined`, as the latter errors if `newToken` isn't defined, but we know in our code that `newToken` is always declared but perhaps not set. It's just good practice to use `typeof newToken !== "undefined"` if what you're checking can actually also be not be declared as a variable yet.